### ISSUE TITLE
Jump to Page does not show entered text under X11

### ISSUE THREAD

**Issue by robertvazan** [zim-desktop-wiki/zim-desktop-wiki#2516]
**Describe the bug**
Since Jump to Page does not work on multi-monitor Wayland (#746), I am using Zim in X11 mode (via X11/Wayland bridge). Some recent update (probably of system software rather than Zim itself since it happened around Fedora 37 -> 39 upgrade) caused text in the edit box in Jump to Page to be invisible. So, under X11 I don't see edit box text and under Wayland I don't see the dropdown window. I am now using the full-text Search as a workaround although that one is far less convenient (could be improved though).

**Environment:**
 - Zim 0.75.2 in flatpak
 - OS: Fedora 39

**Comment by robertvazan** [comment]
PS: I really like the Search window UI - long list inside the window itself rather than a dropdown. I would favor it over Jump to Page in all cases if it supported title-only search, prioritization of prefix matches, search as you type, and quicker keyboard navigation.

**Comment by robertvazan** [comment]
The text in Jump to Page is visible on left-most monitor only, just like in #746 except now the problem is present on X11 too.

After using full-text Search for a while, I have to say that it is not really a satisfactory replacement for Jump to Page. The only way to use Zim productively on multi-monitor linux desktop is to keep it on the left-most monitor. Not great. I am eagerly awaiting the Gtk4 upgrade.

**Comment by prahal** [comment]
From your last comment, there is only one bug ie #746 that affects both X11 and Wayland gtk3. Could you close this X11-only bug report and add the fact the issue also affects Xorg only to #746 if so?
It is hard to follow the progress if the same issue is split into different threads.

Might be good to keep this issue in the zim bug tracker to help people find that it is indeed a gtk3 issue but there is nothing zim devs can do about this issue (out of using another widget to implement this functionality to work around the gtk3 widget bug, or migrate to gtk4 or another framework).

**Comment by robertvazan** [comment]
@prahal I am not sure whether the two bugs are caused by the same underlying issue in Gtk. On X11, editbox text is invisible (except on the leftmost monitor). On Wayland, dropdown window is invisible/misplaced (except on the leftmost monitor). But it probably makes sense to merge the two issues, because they are the same on the level of Zim code. Redesign of the Jump to Page UI would be welcome BTW, because the UI has usability issues in addition to the bugs.
